src/fheroes2/gui/ui_button.cpp

```diff
@@ -955,26 +955,29 @@ namespace fheroes2
         for ( const char * text : texts ) {
             buttonTexts.emplace_back( text, buttonFontType );
         }
+        // This max value is needed to make the width and height calculations correctly take into account that texts with \n are multi-lined.
+        const int32_t maxWidth = 200;
 
         auto maxIter = std::max_element( buttonTexts.begin(), buttonTexts.end(), []( Text & a, Text & b ) { return a.width( a.width() ) < b.width( b.width() ); } );
 
-        const int32_t width = ( *maxIter ).width( ( *maxIter ).width() ) + 6;
+        // We add 6 to have some extra horizontal margin.
+        const int32_t width = ( *maxIter ).width( maxWidth ) + 6;
```

Comment:
We can simplify the access to iterator's method :)
```suggestion
        const int32_t width = maxIter->width( maxWidth ) + 6;
```